CohesionStressRHS.cpp

```diff
@@ -43,35 +48,57 @@ CohesionStressRHS::setDataOnPatch(const int data_idx,
+    Pointer<CellData<NDIM, double>> phi_u_data = patch->getPatchData(d_phi_u_idx);
     Pointer<CellData<NDIM, double>> z_data = patch->getPatchData(d_z_idx);
+    Pointer<CellData<NDIM, double>> w_data = patch->getPatchData(d_w_idx);
+    auto phi_fcn = IBAMR::getKernelAndWidth(d_kernel);
     ret_data->fillAll(0.0);
     if (initial_time) return;
     for (CellIterator<NDIM> i(patch_box); i; i++)
     {
         const CellIndex<NDIM>& idx = i();
         // Compute the ODE terms for stress here
+        // C4 * alpha * I - beta * sigma
+        // compute R2 R3 and R4 -> alpha
+        // where beta = beta(<y>)
+        // where <y> = sqrt(2*Tr(sig) / (S0*z))
+        
+        // Index variables
+        double phi_a = (*phi_a_data)(idx);
+        double phi_u = (*phi_u_data)(idx);
+        double w = (*w_data)(idx);
+        double z = (*z_data)(idx);
+        //const double eta_b = IBAMR::convolution(d_n_b_mx, phi_b, -2.0, z_data, phi_fcn.first, phi_fcn.second, idx, dx);
+        // Compute the source terms
+        const double R2 = d_a0 * phi_a * phi_a;
+        const double R4 = d_a0w * (d_w_mx - w) * phi_u;
+        const double alpha = R2 + R4;
 #if (NDIM == 2)
-        double trace = (*in_data)(idx, 0) + (*in_data)(idx, 1);
-        (*ret_data)(idx, 0) =
-            d_a2 * (*phi_data)(idx) * (*phi_data)(idx)-d_beta_fcn(trace / (*z_data)(idx)) * (*in_data)(idx, 0);
-        (*ret_data)(idx, 1) =
-            d_a2 * (*phi_data)(idx) * (*phi_data)(idx)-d_beta_fcn(trace / (*z_data)(idx)) * (*in_data)(idx, 1);
-        (*ret_data)(idx, 2) = -d_beta_fcn(trace / (*z_data)(idx)) * (*in_data)(idx, 2);
+        const double trace = (*in_data)(idx, 0) + (*in_data)(idx, 1);
+        const double d_y_brackets = std::sqrt(2.0 * trace / z);
+        const double beta = d_beta_fcn(d_y_brackets);
+
+        (*ret_data)(idx, 0) = d_c4 * alpha - beta * (*in_data)(idx, 0);
+        (*ret_data)(idx, 1) = d_c4 * alpha - beta * (*in_data)(idx, 1);
+        (*ret_data)(idx, 2) = -beta * (*in_data)(idx, 2);
 #endif
 #if (NDIM == 3)
-        double trace = (*in_data)(idx, 0) + (*in_data)(idx, 1) + (*in_data)(idx, 2);
-        (*ret_data)(idx, 0) =
-            d_a2 * (*phi_data)(idx) * (*phi_data)(idx)-d_beta_fcn(trace / (*z_data)(idx)) * (*in_data)(idx, 0);
-        (*ret_data)(idx, 1) =
-            d_a2 * (*phi_data)(idx) * (*phi_data)(idx)-d_beta_fcn(trace / (*z_data)(idx)) * (*in_data)(idx, 1);
-        (*ret_data)(idx, 2) =
-            d_a2 * (*phi_data)(idx) * (*phi_data)(idx)-d_beta_fcn(trace / (*z_data)(idx)) * (*in_data)(idx, 2);
-        (*ret_data)(idx, 3) = -d_beta_fcn(trace / (*z_data)(idx)) * (*in_data)(idx, 3);
-        (*ret_data)(idx, 4) = -d_beta_fcn(trace / (*z_data)(idx)) * (*in_data)(idx, 4);
-        (*ret_data)(idx, 5) = -d_beta_fcn(trace / (*z_data)(idx)) * (*in_data)(idx, 5);
+        const double trace = (*in_data)(idx, 0) + (*in_data)(idx, 1) + (*in_data)(idx, 2);
+        const double d_y_brackets = std::sqrt(2.0 * trace / z);
```

Comment:
To prevent dividing by zero:
```suggestion
        const double d_y_brackets = std::sqrt(trace / (z + 1.0e-12));
```